compatibility/allowlist.go

```diff
@@ -0,0 +1,49 @@
+/*
+   Copyright 2020 The Compose Specification Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package compatibility
+
+import (
+	"fmt"
+
+	"github.com/compose-spec/compose-go/errdefs"
+	"github.com/pkg/errors"
+)
+
+// AllowList implement Checker interface by rejecting all attributes but those listed as "supported".
+type AllowList struct {
+	Supported []string
+	errors    []error
+}
+
+func (c *AllowList) Errors() []error {
```

Comment:
Public function missing a comment, we should add a linter to pick things like this up (not in this PR).

```suggestion
// Errors returns the list of errors encountered when checking against the allow list
func (c *AllowList) Errors() []error {
```

Reply:
as this is an interface method, this seems to me a very bad practice to require a comment (until there's some implementation-specific things to say)

Reply:
That makes sense to me. The linter might actually take that into account.